File: Part1-Creating-In-IDE-Course/Ready-Courses/Java-Course/courseSection/courseLesson/programmingTask/src/main/java/org/jetbrains/academy/java/template/InvisibleFile.java
package org.jetbrains.academy.java.template;

// This file will be hidden in the student mode
public class InvisibleFile {
    // This file will be hidden in the student mode
    public static int safeUserInput() {
        String temperature;
        String errorMessage = "The temperature should be in range [-40; 40]";
        do {
            System.out.println("Please input temperature. " + errorMessage);
            temperature = safeReadLine();
        } while (!isCorrectInput(temperature));
        return Integer.parseInt(temperature);
    }

    public static boolean isCorrectInput(String temperature) {
        int intTemp;
        try {
            intTemp = Integer.parseInt(temperature);
        } catch (NumberFormatException e) {
            return false;
        }
        return intTemp > -40 && intTemp < 40;
    }

    public static String safeReadLine() {
        String input;
        try {
            input = System.console().readLine();
        } catch (Exception e) {
            input = null;
        }
        if (input == null) {
            throw new IllegalStateException("Your input is incorrect, sorry. Please, input a number.");
        }
        return input;
    }

}

File: Part1-Creating-In-IDE-Course/Ready-Courses/Kotlin-Course/courseSection/courseLesson/programmingTask/src/main/kotlin/org/jetbrains/academy/kotlin/template/InvisibleFile.kt
package org.jetbrains.academy.kotlin.template

// This file will be hidden in the student mode
fun safeUserInput(): Int {
    var temperature: String
    var isCorrect: Boolean
    val errorMessage = "The temperature should be in range [-40; 40]"
    do {
        println("Please input temperature. $errorMessage:")
        temperature = safeReadLine()
    } while(!isCorrectInput(temperature))
    return temperature.toInt()
}

fun isCorrectInput(temperature: String): Boolean {
    val intTemp = temperature.toIntOrNull() ?: return false
    return intTemp in -40..40
}

fun safeReadLine() = readlnOrNull() ?: error("Your input is incorrect, sorry. Please, input a number.")
